src/features/device-state/routes/state.ts

```diff
@@ -362,24 +483,60 @@ const getConfig = (device: AnyObject) => {
 const getUserAppForState = (
 	device: AnyObject,
 	config: Dictionary<string>,
-): Partial<App> => {
+): AppV3 => {
 	const userAppFromApi: AnyObject = device.belongs_to__application[0];
 
 	// get the release of the main app that this device should run...
 	const release = getReleaseForDevice(device);
 
 	// grab the main app for this device...
 	return release == null
-		? {
+		? ({
 				name: userAppFromApi.app_name,
+				appId: userAppFromApi.id,
+				uuid: userAppFromApi.uuid,
 				services: {},
 				networks: {},
 				volumes: {},
-		  }
+		  } as AppV3)
 		: buildAppFromRelease(device, userAppFromApi, release, config);
 };
 
-const getDependent = (device: AnyObject): Dependent => {
+const getSupervisorAppForState = async (
+	device: any,
+	config: Dictionary<string>,
+): Promise<AppV3 | null> => {
+	if (
+		device.should_be_managed_by__release.length === 0 ||
+		device.should_be_managed_by__release[0].belongs_to__application.length === 0
+	) {
+		return null;
+	}
+
+	const fromApi: AnyObject = {};
+	fromApi.id =
+		device.should_be_managed_by__release[0].belongs_to__application[0].id;
+	fromApi.app_name =
+		device.should_be_managed_by__release[0].belongs_to__application[0].app_name;
+	fromApi.uuid =
+		device.should_be_managed_by__release[0].belongs_to__application[0].uuid;
+	fromApi.application_config_variable =
+		device.should_be_managed_by__release[0].belongs_to__application[0].application_config_variable;
+	fromApi.application_environment_variable =
+		device.should_be_managed_by__release[0].belongs_to__application[0].application_environment_variable;
```

Comment:
Can you please make local references for the release and application? It would really help with readability.

Reply:
Changes made.. resolving since Akis is out.